tests/osn-tests/src/test_osn_volmeter.ts

```diff
@@ -0,0 +1,93 @@
+import 'mocha';
+import { expect } from 'chai';
+import * as osn from 'obs-studio-node';
+import { OBSProcessHandler } from '../util/obs_process_handler';
+
+interface Dictionary<TItemType> {
+    [key: string]: TItemType;
+}
+
+interface IAudioSourceData {
+    fader?: osn.IFader;
+    volmeter?: osn.IVolmeter;
+    callbackInfo?: osn.ICallbackData;
+}
+
+interface IVolmeter {
+    magnitude: number[];
+    peak: number[];
+    inputPeak: number[];
+}
+
+describe('osn-volmeter', () => {
+    let obs: OBSProcessHandler;
+    let inputTypes: string[];
+
+    // Initialize OBS process
+    before(function() {
+        obs = new OBSProcessHandler();
+        
+        if (obs.startup() !== osn.EVideoCodes.Success)
+        {
+            throw new Error("Could not start OBS process. Aborting!")
+        }
+    });
+
+    // Shutdown OBS process
+    after(function() {
+        obs.shutdown();
+        obs = null;
+    });
+
+    context('# Create and Attach', () => {
+        it('Create volmeter and attach it to a audio source', () => {
+            // Creating audio source
+            const input = osn.InputFactory.create('wasapi_input_capture', 'input');
+
+            // Checking if input source was created correctly
+            expect(input).to.not.equal(undefined);
+            expect(input.id).to.equal('wasapi_input_capture');
+            expect(input.name).to.equal('input');
+
+            // Creating volmeter
+            const volmeter = osn.VolmeterFactory.create(osn.EFaderType.IEC);
+
+            // Checking if volmeter was created correctly
+            expect(volmeter).to.not.equal(undefined);
+
+            // Attach volmeter to input source
+            volmeter.attach(input);
```

Comment:
Is there a way we can if this worked?

Reply:
Attach() does not return anything. Right now I can only check if the function is not returning an exception since ValidateResponse throws an exception if something went wrong. If we make the function return something, a boolean for example, I can check the return value after calling it.